### PR TITLE
Rework deterministic sort issue 100

### PR DESCRIPTION
**Pull request by kbuntrock**
see #100

derived from #101

### CODE REVIEW

File: openapi-maven-plugin/src/main/java/io/github/kbuntrock/model/Tag.java [hunk 1]
```diff
@@ -58,4 +66,13 @@ public String toString() {
 			", endpoints=" + endpoints.stream().map(Endpoint::toString).collect(Collectors.joining(", ")) +
 			'}';
 	}
+
+	@Override
+	public int compareTo(final Tag o) {
+		return Comparator
+			.comparing((Tag t) -> t.computedName, nullsLast(String::compareTo))
+			.thenComparing(t -> t.name, nullsLast(String::compareTo))
+			.thenComparing(t -> t.clazz.getCanonicalName())
```
Comment (magx2) [inline openapi-maven-plugin/src/main/java/io/github/kbuntrock/model/Tag.java:75]: I think it makes no sense to sort by clazz name

File: openapi-maven-plugin/src/test/java/io/github/kbuntrock/SpringClassAnalyserTest.java [hunk 2]
```diff
@@ -708,4 +708,33 @@ public void name_collision() throws MojoFailureException, IOException, MojoExecu
...
+
+	@Test
+	public void enpoint_path_collision() throws MojoFailureException, MojoExecutionException {
+
+		final DocumentationMojo mojo = createBasicMojo(
+			io.github.kbuntrock.resources.endpoint.namecollision.three.MyController.class.getCanonicalName());
+
+		MojoRuntimeException exception = null;
+		try {
+			mojo.documentProject();
```
Comment (magx2) [inline openapi-maven-plugin/src/test/java/io/github/kbuntrock/SpringClassAnalyserTest.java:731]: Please use assertj to check for exceptions https://www.baeldung.com/assertj-exception-assertion#1-using-assertthatthrownby

File: openapi-maven-plugin/src/main/java/io/github/kbuntrock/model/OperationType.java [hunk 3]
```diff
@@ -30,7 +30,14 @@ public enum OperationType {
 	private final Class javaxRsVerbAnnotation;
 	private final String jakartaRsVerbAnnotation;
 
-	OperationType(final RequestMethod springMvcRequestMethod, final Class javaxRsVerbAnnotation, final String jakartaRsVerbAnnotation) {
+	/**
+	 * Restitution order in the generated documentation for same urls
+	 */
+	private final int schemaOrder;
```
Comment (magx2) [inline openapi-maven-plugin/src/main/java/io/github/kbuntrock/model/OperationType.java]: You don't need `schemaOrder`, all java enums are comparable by default: https://docs.oracle.com/javase/6/docs/api/java/lang/Enum.html

File: openapi-maven-plugin/src/main/java/io/github/kbuntrock/model/Endpoint.java [hunk 4]
```diff
@@ -96,4 +97,13 @@ public boolean isDeprecated() {
 	public void setDeprecated(final boolean deprecated) {
 		this.deprecated = deprecated;
 	}
+
+	@Override
+	public int compareTo(final Endpoint o) {
+		return Comparator
+			.comparing((Endpoint endpoint) -> endpoint.path, nullsLast(String::compareTo))
+			.thenComparing(e -> e.type.getSchemaOrder())
```
Comment (magx2) [inline openapi-maven-plugin/src/main/java/io/github/kbuntrock/model/Endpoint.java]:
```suggestion
			.thenComparing(e:getType)
```

Enums are comparable by default

**Review by magx2 — CHANGES_REQUESTED** [review]

File: openapi-maven-plugin/src/main/java/io/github/kbuntrock/model/Tag.java (hunk 1 above)
Comment (kbuntrock) [reply openapi-maven-plugin/src/main/java/io/github/kbuntrock/model/Tag.java:75]: in case of name collision, it gives a more deterministic output

Comment (magx2) [reply openapi-maven-plugin/src/main/java/io/github/kbuntrock/model/Tag.java:75]: But the class names are the same

File: openapi-maven-plugin/src/test/java/io/github/kbuntrock/SpringClassAnalyserTest.java (hunk 2 above)
Comment (kbuntrock) [reply openapi-maven-plugin/src/test/java/io/github/kbuntrock/SpringClassAnalyserTest.java:731]: Good call

File: openapi-maven-plugin/src/main/java/io/github/kbuntrock/model/Endpoint.java (hunk 4 above)
Comment (kbuntrock) [reply openapi-maven-plugin/src/main/java/io/github/kbuntrock/model/Endpoint.java]: Oh, nice, removed

**Comment by kbuntrock** [comment]
Fix #100

**Pull request merged**